server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -128,43 +129,52 @@ runHasuraGQ reqId query userInfo resolvedOp = do
   return $ encodeGQResp $ GQSuccess $ encJToLBS resp
 
 -- | Merge the list of response objects by the @data@ key.
-mergeResponseData :: [EncJSON] -> Either String EncJSON
-mergeResponseData responses = do
-  resps <- traverse ((OJ.eitherDecode . encJToLBS) >=> E.parseGQRespValue) responses
-  fmap (OJ.toEncJSON . E.gqrespValueToValue) (mergeGQResp resps)
-
-  where mergeGQResp = flip foldM E.emptyResp $ \accResp E.GQRespValue{..} ->
-          case (gqRespData, gqRespErrors) of
-            (Nothing, Nothing) -> pure accResp
-            (Nothing, Just errors) ->
-              pure
-                accResp
-                  { E.gqRespErrors =
-                      maybe
-                        (Just errors)
-                        (\accErr -> Just $ accErr <> errors)
-                        (E.gqRespErrors accResp)
-                  }
+mergeResponseData :: [EncJSON] -> Bool -> Either String EncJSON
+mergeResponseData responses onlyData = do
+  resps <-
+    traverse ((OJ.eitherDecode . encJToLBS) >=> E.parseGQRespValue) responses
+  fmap
+    (OJ.toEncJSON .
+     (\gqResp ->
+        if onlyData
```

Comment:
I think this change may already fall out from the change I made to gqRespValueToValue here: https://github.com/tirumaraiselvan/graphql-engine/pull/47/commits/b90680c8c55bb4ab2924dcd20d02857645bd4c79#diff-00f8c1aaa790da24e4b812c2d816b02fR94
(from https://github.com/tirumaraiselvan/graphql-engine/pull/47)